consensus/src/liveness/leader_reputation_test.rs

```diff
@@ -141,4 +141,6 @@ fn test_api() {
         &signers[unexpected_index],
     );
     assert!(!proposer_election.is_valid_proposal(&bad_proposal));
+    // Second attempt should fail
+    assert!(!proposer_election.is_valid_proposal(&good_proposal));
```

Comment:
why would sending out a good proposal twice fail? seems a little interesting and not idempotent

Reply:
Same as above^^^